src/dune_rules/menhir/menhir_rules.ml

```diff
@@ -170,6 +173,15 @@ module Run (P : PARAMS) = struct
                       text
                   ]))
 
+  let explain (stanza : stanza) =
+    Ordered_set_lang.Unexpanded.fold_strings stanza.flags ~init:false
+      ~f:(fun _pos sw explain ->
+        explain
+        ||
+        match String_with_vars.text_only sw with
+        | None -> false
+        | Some text -> String.equal "--explain" text)
```

Comment:
No need to go through all the flags once we find one instance.

Reply:
Yes true, but I will have to make an iterator for Ordered_set_lang that exits early. Since the number of flags is small I didn't think it would be a big deal.

Reply:
It's about the intent rather than performance. We're using `fold` where we'd really like to use `find_map`.